community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -131,7 +131,7 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private TransactionHooks.TransactionHooksState hooksState;
     private KernelStatement currentStatement;
```

Comment:
I see what you mean. I double-checked and the original value was 'ANY'. We changed it to 'READ' because it better represented the state of the transaction. The TransactionType is supposed to track when the transaction upgrades in one of two directions, 'data write' and 'schema write'. So the question is what state was it in before this 'upgrade'. We perceived it to be in a 'read' state, meaning that it was not yet performing any kind of write operation. After the upgrade it would be in one of two possible 'write' states.

The older code uses 'ANY' to mark a transaction containing only READ commands, before upgrading to one of the two WRITE states. I think the use of READ makes more sense than ANY. I also think this choice is not related at all to the AccessMode, and makes sense for the TransactionType itself.